lib/rspec/rails/fixture_support.rb

```diff
@@ -2,14 +2,15 @@ module RSpec
   module Rails
     # @private
     module FixtureSupport
-      if defined?(ActiveRecord::TestFixtures)
-        extend ActiveSupport::Concern
-        include RSpec::Rails::SetupAndTeardownAdapter
-        include RSpec::Rails::MinitestLifecycleAdapter if ::ActiveRecord::VERSION::STRING > '4'
-        include RSpec::Rails::MinitestAssertionAdapter
-        include ActiveRecord::TestFixtures
+      extend ActiveSupport::Concern
+
+      included do
+        if RSpec.configuration.use_active_record? && defined?(ActiveRecord::TestFixtures)
+          include RSpec::Rails::SetupAndTeardownAdapter
+          include RSpec::Rails::MinitestLifecycleAdapter if ::ActiveRecord::VERSION::STRING > '4'
+          include RSpec::Rails::MinitestAssertionAdapter
+          include ActiveRecord::TestFixtures
```

Comment:
Moving these includes is undesired

Reply:
I think it's needed due to:

```
An error occurred while loading spec_helper.
Failure/Error: if RSpec.configuration.use_active_record? && defined?(ActiveRecord::TestFixtures)

NoMethodError:
  undefined method `use_active_record?' for #<RSpec::Core::Configuration:0x0000557710919768>
```

when the module looks like this:

```ruby
module FixtureSupport
  if RSpec.configuration.use_active_record? && defined?(ActiveRecord::TestFixtures)
    include RSpec::Rails::SetupAndTeardownAdapter
    include RSpec::Rails::MinitestLifecycleAdapter if ::ActiveRecord::VERSION::STRING > '4'
    include RSpec::Rails::MinitestAssertionAdapter
    include ActiveRecord::TestFixtures    
```

Reply:
Ok so you had a load order issue, you need to make sure the config extension is loaded before this file, then have it like this snippet rather than moving the includes. Moving the includes has the effect of having them all in a different precedence order.

Reply:
Hmm, yeah but that's a chicken egg issue. When I load the config before the fixture_support when it yells at me with:

```
An error occurred while loading spec_helper.
Failure/Error: config.include RSpec::Rails::FixtureSupport, :use_fixtures

NameError:
  uninitialized constant RSpec::Rails::FixtureSupport
```

Reply:
I tried this

```diff
+++ b/lib/rspec/rails/fixture_support.rb
@@ -1,58 +1,57 @@
+require 'rspec/rails/configuration'
 module RSpec
   module Rails
     # @private
     module FixtureSupport
-      extend ActiveSupport::Concern
+      include RSpec::Rails::Configuration
+      if RSpec.configuration.use_active_record? && defined?(ActiveRecord::TestFixtures)
+        include RSpec::Rails::SetupAndTeardownAdapter
+        include RSpec::Rails::MinitestLifecycleAdapter if ::ActiveRecord::VERSION::STRING > '4'
+        include RSpec::Rails::MinitestAssertionAdapter
+        include ActiveRecord::TestFixtures
+        extend ActiveSupport::Concern
```

Without the require because I know I should not do that... 😉
```
An error occurred while loading spec_helper.
Failure/Error: include RSpec::Rails::Configuration

NameError:
  uninitialized constant RSpec::Rails::Configuration
# ./spec/spec_helper.rb:11:in `require'
# ./spec/spec_helper.rb:11:in `<top (required)>'
No examples found.
```

And I have the same error as @Jack12816. Sadly I don't see other way to fix this.

@JonRowe maybe you could tell me what do you mean by
> you need to make sure the config extension is loaded before this file

Reply:
What about if we wrap:
```
config.include RSpec::Rails::FixtureSupport, :use_fixtures
```
In the config if block?

Reply:
I tried

```diff
--- a/spec/spec_helper.rb
+++ b/spec/spec_helper.rb
@@ -23,6 +23,7 @@ RSpec.configure do |config|
   config.filter_run :focus
   config.run_all_when_everything_filtered = true
   config.order = :random
+  config.include RSpec::Rails::FixtureSupport, :use_fixtures
```

And I get
```
An error occurred while loading spec_helper.
Failure/Error: config.include RSpec::Rails::FixtureSupport, :use_fixtures

NameError:
  uninitialized constant RSpec::Rails::FixtureSupport
# ./spec/spec_helper.rb:11:in `require'
# ./spec/spec_helper.rb:11:in `<top (required)>'
No examples found.


Finished in 0.00008 seconds (files took 1.57 seconds to load)
0 examples, 0 failures, 1 error occurred outside of examples
```

`# ./spec/spec_helper.rb:11:in `require'` is `require `rspec/rails'`